src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeEventRepositoryTest.java

```diff
@@ -80,4 +81,46 @@ void should_throw_exception_when_trying_to_create_an_event_for_not_existing_enve
     void should_return_empty_list_when_there_are_no_events_for_given_envelope() {
         assertThat(eventRepo.findForEnvelope(UUID.randomUUID())).isEmpty();
     }
+
+    @Test
+    void should_return_events_for_envelope_ids_order_by_created_at() {
+        // given
+        var envelopeId1 = envelopeRepo.insert(new NewEnvelope("c1", "f1", now(), null, Status.CREATED));
+        var envelopeId2 = envelopeRepo.insert(new NewEnvelope("c2", "f2", now(), null, Status.DISPATCHED));
+
+        var event1a = new NewEnvelopeEvent(envelopeId1, EventType.FILE_PROCESSING_STARTED, null, "note 1");
+        var event2a = new NewEnvelopeEvent(envelopeId2, EventType.FILE_PROCESSING_STARTED, null, "note 3");
+        var event2b = new NewEnvelopeEvent(envelopeId2, EventType.DISPATCHED, null, "note 4");
+
+        // when
+        var eventId1a = eventRepo.insert(event1a);
+        var eventId2a = eventRepo.insert(event2a);
+        var eventId2b = eventRepo.insert(event2b);
+
+        var eventsInDb = eventRepo.findForEnvelopes(asList(envelopeId1, envelopeId2));
+
+        // then
+        assertThat(eventsInDb)
+            .hasSize(3)
+            .usingElementComparatorIgnoringFields("createdAt")
+            .containsExactly(
+                new EnvelopeEvent(eventId2b, envelopeId2, event2b.type, event2b.errorCode, event2b.notes, now()),
```

Comment:
Suggestion, just to consider: we could extract a method, called `envelopeEvent(NewEnvelopeEvent newEnvelopeEvent)` and use it here, referencing passing `event1a`, `event2a` and `event2b` as the only argument. This could be a bit clearer, but it's up to you.

Reply:
Refactored but it needed more params envelope id and event id (auto-generated)